ci: switch to GitHub Actions

barrettk opened the pull request.

barrettk committed:
update pkgr

barrettk committed:
bump roxygen version in description

barrettk committed:
tests helpers: prefer CI env var to drone

barrettk committed:
ci: switch to GitHub Actions

 - coverage display wasn't set up before, so leaving that alone in the readme for now

kyleam commented:
I'm going to remove the review request until you sort out the CI failures, but please let me know if you need any help troubleshooting.

barrettk committed:
update test svgs

kyleam commented on .github/workflows/coverage.yml:
```diff
@@ -0,0 +1,28 @@
+        shell: Rscript {0}
+        env:
+          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
+        run: covr::codecov()
```
`covr::codecov` isn't running successfully.

https://github.com/metrumresearchgroup/pmforest/actions/runs/10289387264/job/28477016357

```
2024-08-07T18:06:58.4091002Z ##[group]Run covr::codecov()
2024-08-07T18:06:58.4091336Z [36;1mcovr::codecov()[0m
2024-08-07T18:06:58.4097627Z shell: /usr/local/bin/Rscript {0}
2024-08-07T18:06:58.4097930Z env:
2024-08-07T18:06:58.4098145Z   MPN_LATEST: 2024-06-12
2024-08-07T18:06:58.4098453Z   R_LIBS_USER: /home/runner/work/_temp/Library
2024-08-07T18:06:58.4098784Z   TZ: UTC
2024-08-07T18:06:58.4099008Z   _R_CHECK_SYSTEM_CLOCK_: FALSE
2024-08-07T18:06:58.4099287Z   NOT_CRAN: true
2024-08-07T18:06:58.4099667Z   RSPM: https://packagemanager.posit.co/cran/__linux__/jammy/latest
2024-08-07T18:06:58.4100317Z   RENV_CONFIG_REPOS_OVERRIDE: https://packagemanager.posit.co/cran/__linux__/jammy/latest
2024-08-07T18:06:58.4100873Z   R_LIB_FOR_PAK: /opt/R/4.4.1/lib/R/site-library
2024-08-07T18:06:58.4101234Z   CODECOV_TOKEN: 
2024-08-07T18:06:58.4101474Z ##[endgroup]
2024-08-07T18:07:14.0761631Z Request failed [429]. Retrying in 1 seconds...
2024-08-07T18:07:15.2792610Z Request failed [429]. Retrying in 3.7 seconds...
2024-08-07T18:07:19.2459084Z $detail
2024-08-07T18:07:19.2460711Z [1] "Rate limit reached. Please upload with the Codecov repository upload token to resolve issue. Expected time to availability: 1924s."
```

Notice that `CODECOV_TOKEN` is empty.  (It'd be marked by asterisks if it were set by a secret.  See bbr's coverage action output.)

So you should either configure that or drop this.  (I think the latter is fine.)

The b57855d message says "coverage display wasn't set up before, so leaving that alone in the readme for now".  I don't follow that, since I don't think the switch to actions would require adjusting the URL.  However, if you don't end up configure this workflow, in my opinion it'd be good to drop the badge because there's no point in leaving a non-functional link in the README.

barrettk replied:
Yeah assumed some additional setup was required to get that working. Im not familiar with the steps and many of our other packages dont run this, so opting to just remove.

kyleam commented on R/helpers-for-tests.R:
```diff
@@ -1,4 +1,4 @@
-#' Skip test if not on Metworx or Drone
+#' Skip test if not on Metworx or CI
```
Hmm, this description seems incorrect to me.  This helper skips if `PMF_SKIP_VDIFFR` is set to true, but the description is "Skip test if not on Metworx or CI".  That's misleading because that environment variable isn't inherently tied to either Metworx or the CI.  And that's demonstrated by the fact that, before this PR, that variable was set in DRONE and this skip was triggered, while the variable is not set in these GitHub Actions and the skip is _not_ triggered.

How about changing this to "Skip test if \`PMF_SKIP_VDIFFR\` is set to true"?

Also, given the description of `skip_vdiffr` says "[t]his is useful because `vdiffr::expect_doppleganger()` is very fragile and tends to fail with false positives in other environments (like CI and `R CMD CHECK`)", should GitHub Actions set this?  Or, given the actions are passing without the skip, should we wait and see how fragile these end up being?

Side remark: the commit title (37eb86b) for this change is "tests helpers: prefer CI env var to drone", but this doesn't look at the CI environment variable.

barrettk replied:
You know I will say I expected the oldest ci action to fail without this variable set. `vdiffr` differences can cause mismatches in svgs, so I find it odd that oldest passed in both [b57855d](https://github.com/metrumresearchgroup/pmforest/pull/37/commits/b57855db30504200d7110d45bf63cd50b61168d4) and [b20ce83](https://github.com/metrumresearchgroup/pmforest/pull/37/commits/b20ce83cc8b67b491a6ac6dc096253074bd567b4).

I'll adjust the documentation, but am leaning towards:

> given the actions are passing without the skip, should we wait and see how fragile these end up being?

kyleam reviewed (changes requested):
Snapshot tests for this branch pass for me on latest Metworx, but fail with main (f518cdb), so I think changes in b20ce83 are good.

```
$ Rscript -e 'devtools::test()'
ℹ Testing pmforest
✔ | F W  S  OK | Context
✔ |         27 | base-plot [11.0s]
✔ |          3 | input-data
✔ |          2 | nsim-plot [1.7s]
✔ |          5 | summary

══ Results ═══════════════════════════════════════════════════════════════════════════════════════════════════════════
Duration: 13.4 s

[ FAIL 0 | WARN 0 | SKIP 0 | PASS 37 ]

$ git rev-parse HEAD
b20ce83cc8b67b491a6ac6dc096253074bd567b4
```

barrettk committed:
remove coverage action and badge from readme

barrettk committed:
adjust documentation for `skip_vdiffr`

 - This environment variable is no longer set in CI. Based on previous findings, this was necessary to avoid differences in SVG/snapshot expectation when installing various versions of `vdiffr` in CI.

 - This doesnt seem to be necessary now. The reasoning is unclear, but opting to _not_ set the environment variable in `.github/workflows/main.yaml` unless necessary.

kyleam reviewed (approved):
Thanks, LGTM
